### PR TITLE
activate C extension

### PR DESCRIPTION
**Pull request by hollow**
please merge the attached patch to activate the C extension. I've also moved around some stuff so that rake-compiler can find all the files. additionally i have moved to SimpleCov for ruby 1.9 compatibility.

### CODE REVIEW

**Comment by codekitchen** [comment]
Thanks. As I stated in #24 I'd rather not require a full C compilation environment just for this one optimization. It appears that this merge request only enables the extension for manual building via the rake task though, and doesn't add it to the gem install, so that's ok with me. Are you planning to manually build a package with the extension for your use?

There are a few other small things to fix up before merging this in, I'll leave inline comments.

File: .rvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+rvm use --create ruby-1.9.3-p374@protocol-buffers
```
Comment (codekitchen) [inline .rvmrc]: please remove this file from the merge, recommend adding it to .gitignore

File: lib/protocol_buffers.rb [hunk 2]
```diff
@@ -1,7 +1,6 @@
-module ProtocolBuffers
-  VERSION = File.read(File.join(File.dirname(__FILE__), "..", "VERSION")).chomp
```
Comment (codekitchen) [inline lib/protocol_buffers.rb:2]: This is a nice change but as written isn't backwards compatible, as now just a `require 'protocol_buffers'` will leave ProtocolBuffers::VERSION undefined. Adding a `require 'protocol_buffers/version'` line to this file will fix that though.

File: ruby-protocol-buffers.gemspec [hunk 3]
```diff
@@ -1,27 +1,28 @@
...
-  s.version     = ProtocolBuffers::VERSION
-  s.platform    = Gem::Platform::RUBY
-  s.authors     = ["Brian Palmer", "Rob Marable", "Paulo Luis Franchini Casaretto"]
-  s.email       = ["brian@codekitchen.net"]
-  s.homepage    = "https://github.com/mozy/ruby-protocol-buffers"
-  s.summary     = %{Ruby compiler and runtime for the google protocol buffers library.}
+Gem::Specification.new do |gem|
+  gem.name          = "ruby-protocol-buffers"
+  gem.version       = ProtocolBuffers::VERSION
+  gem.authors       = ["Brian Palmer", "Rob Marable", "Paulo Luis Franchini Casaretto"]
```
Comment (codekitchen) [inline ruby-protocol-buffers.gemspec]: feel free to add your name here or to the README if you'd like, I've been bad about doing that

**Comment by hollow** [comment]
you are right ... the extension is currently only built with the rake task. this was not intended though. maybe it would make sense to extract varint into its own gem? then it could be optionally installed, just like mongodb is doing it with bson_ext ...

all other things have been fixed :-)

**Comment by codekitchen** [comment]
Yeah that sounds like a good solution to me. I hate ruby gems that print "helpful" suggestions to stdout/stderr on every run though, so I'm going to merge this in and then tweak it. Thanks!

**Comment by hollow** [comment]
thanks! i also made you an owner of the varint gem

**Pull request merged**